core/src/main/java/com/datastrato/gravitino/storage/relation/RelationEntityStore.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.storage.relation;
+
+import static com.datastrato.gravitino.Configs.ENTITY_RELATION_STORE;
+
+import com.datastrato.gravitino.Config;
+import com.datastrato.gravitino.Configs;
+import com.datastrato.gravitino.Entity;
+import com.datastrato.gravitino.EntityAlreadyExistsException;
+import com.datastrato.gravitino.EntitySerDe;
+import com.datastrato.gravitino.EntityStore;
+import com.datastrato.gravitino.HasIdentifier;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.exceptions.AlreadyExistsException;
+import com.datastrato.gravitino.exceptions.NoSuchEntityException;
+import com.datastrato.gravitino.storage.relation.mysql.MySQLBackend;
+import com.datastrato.gravitino.utils.Executable;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Relation store to store entities. This means we can store entities in a relational store. I.e.,
+ * MySQL, PostgreSQL, etc. If you want to use a different backend, you can implement the {@link
+ * RelationBackend} interface
+ */
+public class RelationEntityStore implements EntityStore {
```

Comment:
The `EntityStore` interface seems design for key-value semantics. I'm not sure if we should reuse this interface. If not, there will be more changes. @jerryshao What do you think about this?

Reply:
Can we just have a prototype of how you design this relational entity store interface, as well as a more generic `EntityStore` interface? Then, we can discuss more based on that.

Reply:
I think it is not a big problem to implement it based on the current `EntityStore` interface. The reason is that the actual metadata operation logic is defined in the `Backend`, so it is completely possible to transparently transmit related objects to the `Backend` to customize the logic. I tend to reuse the `EntityStore` interface and reduce related changes.

Reply:
@xloya LGTM.